pkg/apis/serving/v1alpha1/kfservice_validation.go

```diff
@@ -82,10 +80,6 @@ func validateModelSpec(spec *ModelSpec) error {
 	if err := validateReplicas(spec.MinReplicas, spec.MaxReplicas); err != nil {
 		return err
 	}
-	container := spec.CreateModelServingContainer("any", &FrameworksConfig{})
-	if err := knserving.ValidateContainer(*container, sets.String{}); err != nil {
-		return fmt.Errorf("container validation error: %s", err.Error())
-	}
```

Comment:
+1 to delete this awkward code :),  I guess this is not needed since controller creates the container except custom model spec.